homeassistant/components/mazda/lock.py

```diff
@@ -0,0 +1,100 @@
+"""Platform for Mazda lock integration."""
+import datetime
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.const import STATE_LOCKED, STATE_UNLOCKED
+from homeassistant.util import dt as dt_util
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN, LOCK_ASSUMED_STATE_PERIOD
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the lock platform."""
+    client = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+
+    entities = []
+
+    for index, _ in enumerate(coordinator.data):
+        entities.append(MazdaLock(client, coordinator, index))
+
+    async_add_entities(entities)
+
+
+class MazdaLock(MazdaEntity, LockEntity):
+    """Class for the lock."""
+
+    def __init__(self, client, coordinator, index):
+        """Initialize the Mazda entity."""
+        super().__init__(client, coordinator, index)
+        self._assumed_lock_state = None
+        self._assumed_lock_state_timestamp = None
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        vehicle_name = self.get_vehicle_name()
+        return f"{vehicle_name} Lock"
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this entity."""
+        return self.vin
+
+    @property
+    def is_locked(self):
+        """Return true if lock is locked."""
+        api_last_updated_timestamp_str = self.coordinator.data[self.index]["status"][
```

Comment:
`self.coordinator.data[self.index]` appears quite a bit in this integration

It might make sense to make a 
```
    @property
    def data(self):
        """Shortcut to access coordinator data for the entity."""
        return self.coordinator.data[self.index]

```

And then use `self.data`

Reply:
Great idea! I implemented this change.